docs/testing.md

```diff
@@ -0,0 +1,140 @@
+## Testing
+
+`tsaga` comes with its testing utilities.
+
+The utilities are similar to `redux-saga-test-plan` and modeled after how a saga would run in your application.
+
+### Sagas
+
+The assertions use the `expect` function from the `jest` testing framework.
```

Comment:
I don't think that is correct, no implementation is using `jest`. It's just the test runner we have tested by default, but `Ava` for example should work equally fine. (Since in the end we just `throw` on failures).

I think the only place where we import an assertion library is for `deepStrictEqual` from Node's assert.

Reply:
Or are you referring to the below example? Then that sentence should be clarified in that regard.

Reply:
No, in the `expectSaga` and `expectBoundFunction` we are using the `expect` global function from `jest`. I think they also publish it as an npm module which we should probably use.

Reply:
Oh, missed that because we also use `deepStrictEqual` in the old test helper.

Hmm, let's see, does `expect` have a much better output? Otherwise I would suggest switching to `deepStrictEqual`. (If `expect` is better, maybe we can use the underlying library directly, and skip a full `jest` dependency.)

This seems really bad overall though, since the only reason why it's working is that we by accident also use `jest` as a test runner and that then provides the global expect.

I wonder how that even works, because we don't have `jest` in the `tsconfig.json` (unlike the AYSA apps, which do have that IIRC).

Reply:
It works because we have the jest types installed. I’m not sure, but I think the expected/received output comes from the expected function.